code/datums/riding.dm

```diff
@@ -252,31 +252,195 @@
+			ridden.pixel_x = -16
+			ridden.pixel_y = -16
+		if(EAST,WEST)
+			ridden.pixel_x = -18
+			ridden.pixel_y = 0
+	if(turret)
+		if(ridden.dir == SOUTH)
+			turret.layer = 4
+		else
+			turret.layer = 5
+
+/datum/riding/space/repair/handle_vehicle_offsets()
+	..()
+	R = ridden
+	turret = R.turret
+	if(turret)
+		turret.forceMove(get_turf(ridden))
+		switch(ridden.dir)
+			if(NORTH)
+				turret.pixel_x = 1
+				turret.pixel_y = -16
+			if(EAST)
+				turret.pixel_x = -23
+				turret.pixel_y = 10
+			if(SOUTH)
+				turret.pixel_x = 1
+				turret.pixel_y = 23
+			if(WEST)
+				turret.pixel_x = 23
+				turret.pixel_y = 10
+	if(ridden.has_buckled_mobs())
+		for(var/m in ridden.buckled_mobs)
+			var/mob/living/buckled_mob = m
+			buckled_mob.setDir(ridden.dir)
+			switch(ridden.dir)
+				if(NORTH)
+					buckled_mob.pixel_x = 0
+					buckled_mob.pixel_y = -8
+				if(SOUTH)
+					buckled_mob.pixel_x = 0
+					buckled_mob.pixel_y = 4
+				if(EAST)
+					buckled_mob.pixel_x = -10
+					buckled_mob.pixel_y = 5
+				if(WEST)
+					buckled_mob.pixel_x = 10
+					buckled_mob.pixel_y = 5
+
+//ATMOBILE
+
+datum/riding/space/speedbike/atmos
+	vehicle_move_delay = 1
+
+
+//ENGIWAGON
+
+/datum/riding/space/speedbike/engiwagon/handle_vehicle_offsets()
+	var/seat_number = 0
+	if(ridden.has_buckled_mobs())
+		for(var/m in ridden.buckled_mobs)
+			var/mob/living/buckled_mob = m
+			seat_number++
+			buckled_mob.setDir(ridden.dir)
+			if(seat_number == 1)
+				switch(ridden.dir)
+					if(NORTH)
+						buckled_mob.pixel_x = -8
+						buckled_mob.pixel_y = -5
+					if(SOUTH)
+						buckled_mob.pixel_x = 9
+						buckled_mob.pixel_y = -1
+					if(EAST)
+						buckled_mob.pixel_x = -4
+						buckled_mob.pixel_y = 4
+						buckled_mob.layer = 4
+					if(WEST)
+						buckled_mob.pixel_x = 8
+						buckled_mob.pixel_y = 5
+						buckled_mob.layer = 4.1
+			if(seat_number == 2)
+				switch(ridden.dir)
+					if(NORTH)
+						buckled_mob.pixel_x = 9
+						buckled_mob.pixel_y = -5
+					if(SOUTH)
+						buckled_mob.pixel_x = -8
+						buckled_mob.pixel_y = -1
+					if(EAST)
+						buckled_mob.pixel_x = -8
+						buckled_mob.pixel_y = 5
+						buckled_mob.layer = 4.1
+					if(WEST)
+						buckled_mob.pixel_x = 4
+						buckled_mob.pixel_y = 4
+						buckled_mob.layer = 4
+
 //SPEEDUWAGON
 
 /datum/riding/space/speedwagon
 	vehicle_move_delay = 0
 
 /datum/riding/space/speedwagon/handle_vehicle_offsets()
+	var/seat_number = 0
 	if(ridden.has_buckled_mobs())
 		for(var/m in ridden.buckled_mobs)
 			var/mob/living/buckled_mob = m
+			seat_number++
 			buckled_mob.setDir(ridden.dir)
 			ridden.pixel_x = -48
 			ridden.pixel_y = -48
-			switch(ridden.dir)
-				if(NORTH)
-					buckled_mob.pixel_x = -10
-					buckled_mob.pixel_y = -3
-				if(SOUTH)
-					buckled_mob.pixel_x = 16
-					buckled_mob.pixel_y = 3
-				if(EAST)
-					buckled_mob.pixel_x = -4
-					buckled_mob.pixel_y = 30
-				if(WEST)
-					buckled_mob.pixel_x = 4
-					buckled_mob.pixel_y = -1
+			if(seat_number == 1)
```

Comment:
p sure you can just use buckled_mob[1] [2] or whatever the syntax was ask antur or something